concrete/controllers/single_page/dashboard/system/attributes/topics.php

```diff
@@ -100,4 +100,36 @@ public function remove_tree_node()
             Loader::helper('ajax')->sendError($this->error);
         }
     }
+
+    public function tree_edit()
+    {
+        $vs = Loader::helper('validation/strings');
+
+        if (!$vs->notempty($_REQUEST['treeName'])) {
```

Comment:
Instead of accessing the global `$_REQUEST` variable, you should use the `Request` class:
- in order to get querystring variables, use `$this->request->query->get('treeName');` instead of `$_POST['treeName']`
- in order to get posted variables, use `$this->request->request->get('treeName');` instead of `$_GET['treeName']`
- the closest replacement of `$_REQUEST['treeName']` could be `$this->request->get('treeName')`, but unfortunately it's a breaking change, since with `$_REQUEST` the posted variables have a higher precedence over the querystring variables, whereas with `$this->request->get()` it's the opposite.  
So, the safest approach is  
`$treeName = $this->request->request->get('treeName', $this->request->query->get('treeName'));`